pkg/controller/job/job_controller.go

```diff
@@ -812,21 +813,15 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 		klog.V(4).InfoS("Too many pods running for job", "job", klog.KObj(job), "deleted", len(podsToDelete), "target", parallelism)
 		removed, err := jm.deleteJobPods(job, jobKey, podsToDelete)
 		active -= removed
-		if err != nil {
-			return active, err
-		}
+		// While it is possible for a Job to require both pod creations and
+		// deletions at the same time (e.g. indexed Jobs with repeated indexes), we
+		// restrict ourselves to either just pod deletion or pod creation in any
+		// given sync cycle. Of these two, pod deletion takes precedence.
+		return active, metrics.JobSyncActionPodsDeleted, err
 	}
 
 	if active < wantActive {
 		diff := wantActive - active
-		if diff < 0 {
```

Comment:
Any particular reason to drop this safety net?

Reply:
but it's already checked in line 820

Reply:
because of the condition above, `wantActive` is strictly larger than `active`

Reply:
ha, I should have refreshed my page, I didn't see your replies :)